lib/Importer/ONNXModelLoader.cpp

```diff
@@ -256,6 +256,96 @@ bool ONNXModelLoader::loadOperator(const ONNX_NAMESPACE::NodeProto &op) {
     return true;
   }
 
+  if (typeName == "Slice") {
+    auto data = getNodeValueOrCreateConstantByName(op.input(0));
+    auto dims = data.dims();
+    auto numDims = dims.size();
+
+    // Attributes 'starts' and 'ends' are mandatory and must be consistent.
+    assert(dict.count("starts") && "Slice: attribute 'starts' is mandatory.");
+    assert(dict.count("ends") && "Slice: attribute 'ends' is mandatory.");
+    auto starts = getShape<ssize_t>(dict["starts"]);
+    auto ends = getShape<ssize_t>(dict["ends"]);
+    assert((starts.size() == ends.size()) &&
+           "Slice: 'starts' and 'end' arrays must have the same size.");
+
+    // Attribute 'axes' is optional.
+    std::vector<ssize_t> axes;
+    if (dict.count("axes")) {
+      // The ONNX spec is unclear so we consider that the 'axes' array may have
+      // any size. The constraints are:
+      // - the element value must be in range [0, numDims[,
+      // - 'starts' & 'ends' arrays must have the same size as the 'axes' array.
+      // In case an axe is specified multiple times, the later parameters will
+      // simply overwrite the previous one.
+      axes = getShape<ssize_t>(dict["axes"]);
+      assert((axes.size() == starts.size()) &&
+             "Slice: The 'axes' array must have the same size as 'starts' and "
+             "'ends' arrays.");
+    } else {
+      for (size_t i = 0; i < numDims; i++) {
+        axes.push_back(ssize_t(i));
+      }
+    }
+
+    // The ONNX description is unclear and doesn't describe what to do when a a
+    // sub-set of axes index are missing in the axes array. An interpretation
+    // is that in this case, the entire range is considered for the non
+    // specified axes. Then, initialize newStarts and newEnds with the full
+    // index range for all axes.
+    std::vector<size_t> newStarts(numDims), newEnds(numDims);
+    for (size_t i = 0; i < numDims; i++) {
+      newStarts[i] = 0;
+      newEnds[i] = dims[i];
+    }
+
+    // Determine the coordinates of the sub-tensor to extract.
+    assert(starts.size() == ends.size());
+    for (size_t i = 0; i < axes.size(); i++) {
+      ssize_t newStart = starts[i];
+      ssize_t newEnd = ends[i];
+      ssize_t axesId = axes[i];
```

Comment:
nit maybe this should be `axisId` if it's just the one

Reply:
Done